src/SourceLocator/Type/StringSourceLocator.php

```diff
@@ -25,9 +27,9 @@ class StringSourceLocator extends AbstractSourceLocator
      * @param string $source
      * @throws \Roave\BetterReflection\SourceLocator\Exception\EmptyPhpSourceCode
      */
-    public function __construct(string $source)
+    public function __construct(string $source, ?Locator $locator = null)
```

Comment:
IMO, the dependency should be the parser, and the dependency shouldn't be optional.